api/appointmentsPOST.ts

```diff
@@ -0,0 +1,115 @@
+import handler from './handler';
+import db from './db';
+import { QueryResult } from 'pg';
+
+export default handler(async (request: any) => {
+  
+  return await db(async (connection) => {
+    // Take data from request and build sql call from it
+  // TODO: Move this data structure to it's own class
+  const {
+    //Sharded Values
+    patientid,
+    //TFI Survey
+    tfi_i,
+    tfi_sc,
+    tfi_c,
+    tfi_si,
+    tfi_a,
+    tfi_r,
+    tfi_q,
+    tfi_e,
+    tfi_overallscore,
+    //THS Survey
+    ths_sectiona,
+    ths_sectionb,
+    ths_sectionc,
+    ths_sectionc_example,
+    //TS Survey
+    ts_type,
+    //Appointment
+    authorityid,
+    //tfisurveyid,
+    //thssurveyid,
+    //tssurveyid,
+    appointmentdatetime,
+    tympanometrytype, 
+    otoscopytype, 
+    rightear_lowf_severity, 
+    rightear_highf_severity, 
+    leftear_lowf_severity, 
+    leftear_highf_severity, rightear_lowf_configuration, rightear_highf_configuration, leftear_lowf_configuration, leftear_highf_configuration, audiogramtype
+  } = request.body;
+
+  let tfi_datapoints : string[]  = [
+    'patientid',
+    'tfi_i',
+    'tfi_sc',
+    'tfi_c',
+    'tfi_si',
+    'tfi_a',
+    'tfi_r',
+    'tfi_q',
+    'tfi_e',
+    'tfi_overallscore'
+  ];
+
+  let ths_datapoints : string[] = [
+    'patientid',
+    'ths_sectiona',
+    'ths_sectionb',
+    'ths_sectionc',
+    'ths_sectionc_example'  
+  ];
+
+  let ts_datapoints : string[] = [
+    'patientid',
+    'ts_type'
+  ]
+
+  let audiologistexams_datapoints : string[] = [
+    'tympanometrytype', 
+    'otoscopytype', 
+    'rightear_lowf_severity', 
+    'rightear_highf_severity', 
+    'leftear_lowf_severity', 
+    'leftear_highf_severity',
+    'rightear_lowf_configuration',
+    'rightear_highf_configuration',
+    'leftear_lowf_configuration',
+    'leftear_highf_configuration', 
+    'audiogramtype'
+  ]
+
+  let tfisurvey_sql : string = "INSERT INTO tfisurvey (" + tfi_datapoints.join(", ") + ") VALUES ("+ tfi_datapoints.map((value, index) => "$" + (index + 1)) + ") RETURNING *";  
+  let tfisurvey_values = [patientid, tfi_i, tfi_sc, tfi_c, tfi_si, tfi_a, tfi_r, tfi_q, tfi_e, tfi_overallscore];
```

Comment:
You can use `map` to transform `tfi_datapoints` into this, by turning `name` into `request.body[name]`